test/conformance/ingress/percentage_test.go

```diff
@@ -0,0 +1,127 @@
+	const headerName = "Foo-Bar-Baz"
+
+	backends := make([]v1alpha1.IngressBackendSplit, 0, 10)
+	weights := make(map[string]float64, len(backends))
+
+	// Double the percentage of the split each iteration until it would overflow, and then
+	// give the last route the remainder.
+	percent, total := 1, 0
+	for i := 0; i < 10; i++ {
+		name, port, cancel := CreateService(t, clients, networking.ServicePortNameHTTP1)
+		defer cancel()
+		backends = append(backends, v1alpha1.IngressBackendSplit{
+			IngressBackend: v1alpha1.IngressBackend{
+				ServiceName:      name,
+				ServiceNamespace: test.ServingNamespace,
+				ServicePort:      intstr.FromInt(port),
+			},
+			// Append different headers to each split, which lets us identify
+			// which backend we hit.
+			AppendHeaders: map[string]string{
+				headerName: name,
+			},
+			Percent: percent,
+		})
+		weights[name] = float64(percent)
+
+		total += percent
+		percent *= 2
+		// Cap the final non-zero bucket so that we total 100%
+		// After that, this will zero out remaining buckets.
+		if total+percent > 100 {
+			percent = 100 - total
+		}
+	}
+
+	// Create a simple Ingress over the 10 Services.
+	name := test.ObjectNameForTest(t)
+	_, client, cancel := CreateIngressReady(t, clients, v1alpha1.IngressSpec{
+		Rules: []v1alpha1.IngressRule{{
+			Hosts:      []string{name + ".example.com"},
+			Visibility: v1alpha1.IngressVisibilityExternalIP,
+			HTTP: &v1alpha1.HTTPIngressRuleValue{
+				Paths: []v1alpha1.HTTPIngressPath{{
+					Splits: backends,
+				}},
+			},
+		}},
+	})
+	defer cancel()
+
+	// Create a large enough population of requests that we can reasonably assess how
+	// well the Ingress respected the percentage split.
+	seen := make(map[string]int, len(backends))
+	const totalRequests = 1000
+	for i := 0; i < totalRequests; i++ {
+		// Make a request and check the response.
+		resp, err := client.Get("http://" + name + ".example.com")
+		if err != nil {
+			t.Fatalf("Error making GET request: %v", err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			t.Errorf("Got non-OK status: %d", resp.StatusCode)
+			continue
+		}
+
+		b, err := ioutil.ReadAll(resp.Body)
+		ri := &types.RuntimeInfo{}
+		if err := json.Unmarshal(b, ri); err != nil {
+			t.Fatalf("Unable to parse runtime image's response payload: %v", err)
+		}
+		// Increment the bucket for the seen header.
+		seen[ri.Request.Headers.Get(headerName)]++
+	}
+
+	const margin = 2.0 // Allow the Ingress to be within 2% of the configured value.
+	for name, want := range weights {
+		got := (float64(seen[name]) * 100.0) / float64(totalRequests)
+		switch {
+		case want == 0.0 && got > 0.0:
+			// For 0% targets, we have tighter requirements.
+			t.Errorf("target %q received traffic, wanted none (0%% target).", name)
+		case want-margin > got || got > want+margin:
+			t.Errorf("target %q received %f%%, wanted %f +/- %f", name, got, want, margin)
```

Comment:
```suggestion
			t.Errorf("Target %q received %f%%, wanted %f +/- %f", name, got, want, margin)
```
I also like the way I did report there: `"want in (%f, %f) interval"..., want-margin, want+margin)`